src/server/job_func.c

```diff
@@ -775,7 +775,7 @@ job_purge(job *pjob)
 		(pjob->ji_qs.ji_substate != JOB_SUBSTATE_TRANSICM)) {
 		if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
 			svr_dequejob(pjob);
-			if (pjob->ji_qs.ji_substate == JOB_SUBSTATE_RERUN3)
+			if ((pjob->ji_qs.ji_substate == JOB_SUBSTATE_RERUN3) || (pjob->ji_qs.ji_substate == JOB_SUBSTATE_QUEUED))
```

Comment:
why is this check for queued substate needed?

Reply:
@arungrover it's needed becasue before calling call_to_process_hooks, create_subjob() changed the state(JOB_STATE_QUEUED to JOB_SUBSTATE_QUEUED) but in job_purge() we were checking for JOB_SUBSTATE_RERUN3 state only.